src/main/io/serial.c

```diff
@@ -211,11 +212,13 @@ bool isSerialConfigValid(serialConfig_t *serialConfigToCheck)
                 return false;
             }
 
-            if (!(portConfig->functionMask & FUNCTION_MSP)) {
```

Comment:
You must handle 2 cases here:
- MSP is set, check second function
- Blackbox is set, check second function (serialrx)

Also only some type of serial RX is shareable